python/pyarrow/pandas_compat.py

```diff
@@ -191,18 +193,61 @@ def construct_metadata(df, column_names, index_levels, preserve_index, types):
             for i, (level, arrow_type) in enumerate(zip(index_levels,
                                                         index_types))
         ]
+        column_index_names = list(df.columns.names)
+        column_index_dtypes = [
+            str(level.dtype) for level in getattr(
```

Comment:
Does this dtype need to be stored in the same way that we're storing normal pandas column dtypes (with logical + physical)?  So maybe this will be a dict